### ISSUE TITLE
Login form produces "Index is out of range" in some cases

### ISSUE THREAD

**Issue by roman-yagodin** [DNNCommunity/DNN.ActiveDirectory#17]
```
2017-08-28 11:48:17,329 [web][Thread:122][FATAL] DotNetNuke.Framework.PageBase - An error has occurred while loading page.
System.IndexOutOfRangeException: Индекс находился вне границ массива.
   в DotNetNuke.Authentication.ActiveDirectory.Login.cmdLogin_Click(Object sender, EventArgs e)
   в System.Web.UI.WebControls.LinkButton.OnClick(EventArgs e)
   в System.Web.UI.WebControls.LinkButton.RaisePostBackEvent(String eventArgument)
   в System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)
2017-08-28 11:48:17,345 [web][Thread:122][ERROR] DotNetNuke.Services.Exceptions.Exceptions - ~/Default.aspx?tabid=1641&error=An unexpected error has occurred
System.IndexOutOfRangeException: Индекс находился вне границ массива.
   в DotNetNuke.Authentication.ActiveDirectory.Login.cmdLogin_Click(Object sender, EventArgs e)
   в System.Web.UI.WebControls.LinkButton.OnClick(EventArgs e)
   в System.Web.UI.WebControls.LinkButton.RaisePostBackEvent(String eventArgument)
   в System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)
```

**Comment by sawest** [comment]
Is this the entire stack?  I don't see an "Index is out of bounds" in the stack? If its no the entire stack please post the entire stack.

You state it happens in some cases, do you know in what cases it occurs?  How can I duplicate the problem?

**Comment by roman-yagodin** [comment]
I've updated stack trace (some messages in russian). I do not know how to reproduce the issue, it just started after some time of trying to login and persisted after that.

**Comment by sawest** [comment]
what is the username you are trying to use?  You dont have to specify the exact username but I need to know the exact format....e.g. username, or domain\username or username@domain.com, etc?

**Comment by roman-yagodin** [comment]
If I remember correctly, I used domain\username format.

**Comment by sawest** [comment]
Found a bug that could cause this.  Pull request will be coming shortly.

**Comment by sawest** [comment]
fixed in pull request #27
